cmd/initialise.go

```diff
@@ -0,0 +1,190 @@
+		// make vault init request and get root token
+		fmt.Println("Initialising Vault...")
+		rootToken := initVault()
+		fmt.Println("Initialisation complete")
+
+		// AWS setup
+		region := os.Getenv("AWS_REGION")
+		sess := session.Must(session.NewSession(&aws.Config{Region: aws.String(region)}))
+		var kmsClient = kms.New(sess, aws.NewConfig().WithRegion(region))
+		uploader := s3manager.NewUploader(sess)
+
+		// encrypt tokens with AWS KMS
+		fmt.Println("Encrypting root token...")
+		encryptedToken, errE := kmsClient.Encrypt(&kms.EncryptInput{
+			KeyId: aws.String(fullKeyID(os.Args[1], os.Args[2], region)),
+			Plaintext: []byte(rootToken),
+		})
+		checkError(errE)
+		fmt.Println("Encryption complete.")
+
+		hostname, errH := os.Hostname()
+		if errH != nil {
+			panic(errH)
+		}
+
+		tokenFileName := hostname+ "_token"
+
+		writeToFile(tokenFileName, encryptedToken.CiphertextBlob)
+
+		// upload keys to S3
+		fmt.Println("Uploading encrypted token to S3...")
+		f := openFile(tokenFileName)
+
+		s3Result, errS3 := uploader.Upload(&s3manager.UploadInput{
+			Bucket: aws.String("encrypted-tokens"),
+			Key:    aws.String(tokenFileName),
+			Body:   f,
+		})
+		checkError(errS3)
+		fmt.Println("Encrypted token successfully uploaded to S3 at", s3Result.Location)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(initialiseCmd)
+}
+
+func checkError(e error) {
+	if e != nil {
+		panic(e)
+	}
+}
+
+func initVault() string {
+	vaultAddr := os.Getenv("VAULT_ADDR")
+	if vaultAddr == "" {
+		vaultAddr = "http://127.0.0.1:8200"
+	}
+
+	httpClient = http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: false,
+			},
+		},
+	}
+
+	initRequest := InitPayload{
+		RecoveryShares:    1,
+		RecoveryThreshold: 1,
+		SecretShares:      5,
+		SecretThreshold:   3,
+	}
+
+	initRequestData, errMash := json.Marshal(&initRequest)
+	checkError(errMash)
+
+	reader := bytes.NewReader(initRequestData)
+	request, errR := http.NewRequest("PUT", vaultAddr+"/v1/sys/init", reader)
+	checkError(errR)
+
+	response, errReq := httpClient.Do(request)
+	checkError(errReq)
+
+	defer response.Body.Close()
+
+	initRequestResponseBody, errResp := ioutil.ReadAll(response.Body)
+	checkError(errResp)
+
+	if response.StatusCode != 200 {
+		log.Printf("init: non 200 status code: %d", response.StatusCode)
+		return string(response.StatusCode)
+	}
+
+	var initResponse InitResponse
+
+	if errUn := json.Unmarshal(initRequestResponseBody, &initResponse); errUn != nil {
+		log.Println(errUn)
+		panic(errUn)
+	}
+
+
+	return initResponse.RootToken
+}
+
+func writeToFile(filename string, content []byte) {
+    err := ioutil.WriteFile(filename, content, 0644)
+    checkError(err)
+}
+
+func openFile(filename string) *os.File {
+	f, errF  := os.Open(filename)
+	checkError(errF)
+
+	return f
+}
+
+func fullKeyID(accountID string, keyID string, region string) (string) {
+	baseString := fmt.Sprintf("arn:aws:kms:region:%s:key/%s", accountID, keyID)
```

Comment:
Looking at the function signature, do we need to replace `region` here?

Reply:
Yes, you're right.